src/components/CharactersList.tsx

```diff
@@ -0,0 +1,70 @@
+import { useQuery } from "@apollo/client";
+import { GET_CHARACTERS } from "../graphql/Characters";
+import CharacterCard from "./CharacterCard";
+import { CharacterListProps, CharactersData } from "../utils/types";
+import { useEffect } from "react";
+import Loading from "./Loading";
+
+const CharactersList: React.FC<CharacterListProps> = ({
+  handleSelectedCharacter,
+}) => {
+  const { loading, data, fetchMore } = useQuery<CharactersData>(
+    GET_CHARACTERS,
+    {
+      variables: {
+        page: 1,
+      },
+      notifyOnNetworkStatusChange: true,
+    }
+  );
+
+  const characters = data?.characters?.results || [];
+  const nextPage = data?.characters?.info?.next;
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const handleScroll = () => {
```

Comment:
Sadly this doesn't work properly, how ever the fetchMore function was correct

Reply:
try to iterate your solution using this package https://www.npmjs.com/package/react-infinite-scroll-component